### PR TITLE
feat: shared packages

### PR DESCRIPTION
**Pull request by cbrzn**
Introduces the new structure of the wraps & crates, where two wraps have been created:
- Core: 
   - Has the **same** functionality as the previous Ethereum wrapper, except for the following methods:
      - `encodeFunction`
      - `encodeParams`
      - `toEth`
      - `toWei`
   - The `*_sync` string has been removed from methods & file, to clean the codebase and make it more maintainable
 
- Utils:
    - This wrap allows users to interact with Ethereum utility functions. The methods removed from `Core` were moved here
    - Doesn't need the `provider` dependency
    - `solidityPack` was implemented, being compatible with `ethers.js` implementation :smile: 

- Also updates `ethers` version to `2.0.2`

This is the first iteration of the new structure. This will have two more iterations which will be tracked by these issues (in order):
- https://github.com/polywrap/ethereum/issues/86
- https://github.com/polywrap/ethereum/issues/85
     
Also, I wanted to propose that once we merge this, we deploy these new wrappers to the ENS domains:
- core -> `ethers.wraps.eth:core@0.0.1`
- utils -> `ethers.wraps.eth:utils@0.0.1`

We don't have an easy way to export this two wrappers through the same interface because of this error in the schema compose: https://github.com/polywrap/toolchain/issues/1731 - So I think it's okay to start at least using this `core` and `utils` wraps from ENS. Open to suggestions tho

So we can start migrating uniswap and safe wrappers to this structure and battle test these news wraps :stuck_out_tongue:

### CODE REVIEW

File: .github/workflows/wrapper-ci.yaml [hunk 1]
```diff
@@ -1,15 +1,14 @@
-name: ethereum-wrapper-ci
+name: ethers-wrappers-ci
```
Comment (krisbitney) [inline .github/workflows/wrapper-ci.yaml:1]: Should the utils and core wraps each have their own CI?

File: wraps/core/src/api/transaction.rs [hunk 2]
```diff
@@ -1,155 +1,171 @@
...
             return Err(WrapperError::ContractError(error));
-        },
+        }
         (None, true) => bytecode.clone(),
         (Some(constructor), _) => {
-            let tokens: Vec<Token> = tokenize_values(&values, &constructor.inputs);
-            constructor.encode_input(bytecode.to_vec(), &tokens)?.into()
-        },
+            let tokens: Vec<Token> = ethers_utils::tokenize_values(&values, &constructor.inputs);
+            //TODO (cbrzn): Remove unwrap
```
Comment (krisbitney) [inline wraps/core/src/api/transaction.rs:76]: Could you make this TODO statement clearer so it's obvious what needs to be done? I don't quite understand what you mean here

**Review by krisbitney — APPROVED** [review]
This is really great work. Thanks for everything you've done to get us here. 

I added a couple of comments but they are really not important. Feel free to ignore them.

**Review by dOrgJelli — APPROVED** [review]
This all LGTM, aside from:
- fleshing out the README of each wrap
- adding comments to the core wrap's interface (I used ChatGPT for the utils interface, helped speed it up a bit)
- improving typings & variable names on various functions (ex: `toWei(eth: String!)` -> `toWei(eth: BigInt!)`)

File: wraps/core/src/api/transaction.rs (hunk 2 above)
Comment (cbrzn) [reply wraps/core/src/api/transaction.rs:76]: woops, nice catch

**Pull request merged**